core/eolearn/core/eodata.py

```diff
@@ -83,11 +71,11 @@ def __setattr__(self, key, value, feature_name=None):
 
         In case they are a dictionary they are cast to _FeatureDict class
         """
-        if feature_name is not None and FeatureType.has_value(key):
+        if feature_name not in (None, Ellipsis) and FeatureType.has_value(key):
```

Comment:
This covers both options - what a user might provide and what a `FeatureParser` might provide. For now, I would leave it as it is but I agree that it is a bit confusing and that sometime in the future it could be improved. But that would require quite some time.